streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -91,9 +91,9 @@ public void process(final K key, final V value) {
             final String keyClass = key == null ? "unknown because key is null" : key.getClass().getName();
             final String valueClass = value == null ? "unknown because value is null" : value.getClass().getName();
             throw new StreamsException(
-                    String.format("A serializer (key: %s / value: %s) is not compatible to the actual key or value type " +
+                    String.format("ClassCastException while producing data to a sink topic. A serializer (key: %s / value: %s) is not compatible to the actual key or value type " +
                                     "(key type: %s / value type: %s). Change the default Serdes in StreamConfig or " +
-                                    "provide correct Serdes via method parameters.",
+                                    "provide correct Serdes via method parameters (for example if using the DSL, `#to(String topic, Produced<K, V> produced)` with `Produced.keySerde(WindowedSerdes.timeWindowedSerdeFrom(String.class))`).",
```

Comment:
Add unit test for this?

Reply:
Not sure what you mean?

Reply:
Sorry, I was trying to ask if I should write a unit test to capture the scenarios I was describing in the highlighted code - a test to see if the exception is used or not when using Produced.keySerde method.

Reply:
I think the existing test are sufficient.